test/unit/io/OGRWriterTest.cpp

```diff
@@ -0,0 +1,436 @@
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include <pdal/pdal_test_main.hpp>
+#include <pdal/util/FileUtils.hpp>
+#include <pdal/private/gdal/ErrorHandler.hpp>
+#include <filters/HeadFilter.hpp>
+#include <filters/RangeFilter.hpp>
+#include <filters/SortFilter.hpp>
+#include <io/BufferReader.hpp>
+#include <io/FauxReader.hpp>
+#include <io/OGRWriter.hpp>
+#include <io/LasReader.hpp>
+#include "Support.hpp"
+
+#include <iostream>
+#include <sstream>
+
+namespace pdal
+{
+
+namespace
+{
+
+void runOgrWriterInfo(const Options& wo, const std::string& infile,
+    const std::string& infofile, const std::string suffix, int featureCount = 10,
+    uint32_t(*compare)(const std::string&, const std::string&, int32_t) = &Support::diff_text_files
+)
+{
+    FileUtils::createDirectory(Support::temppath("ogr"));
+    std::string outfile(Support::temppath(std::string("ogr/") + FileUtils::stem(FileUtils::stem(FileUtils::getFilename(infofile))) + suffix));
+    std::string outinfofile(outfile + ".ogrinfo");
+
+    if(FileUtils::isDirectory(outfile)) {
+        FileUtils::deleteDirectory(outfile);
+    } else {
+        FileUtils::deleteFile(outfile);
+    }
+    FileUtils::deleteFile(outinfofile);
+
+    Options readerOpts;
+    readerOpts.add("filename", infile);
+
+    LasReader reader;
+    reader.setOptions(readerOpts);
+
+    Options sortOpts;
+    SortFilter sortFilter;
+    sortOpts.add("dimension", "X");
+    sortFilter.setOptions(sortOpts);
+    sortFilter.setInput(reader);
+
+    Options writerOpts;
+    writerOpts.add("filename", outfile);
+
+    OGRWriter writer;
+    writer.setOptions(wo);
+    writer.addOptions(writerOpts);
+
+    HeadFilter headFilter;
+    if (featureCount > 0) {
+        Options headOpts;
+        headOpts.add("count", featureCount);
+        headFilter.setOptions(headOpts);
+        headFilter.setInput(sortFilter);
+        writer.setInput(headFilter);
+    } else {
+        writer.setInput(sortFilter);
+    }
+
+    PointTable t;
+    writer.prepare(t);
+    writer.execute(t);
+
+    std::string cmd = "ogrinfo -nomd -al " + outfile;
+    if (featureCount == 0) {
+        cmd += " -so";
+    }
+    cmd += " 2>&1";
+    std::string info;
+    if (Utils::run_shell_command(cmd, info)) {
+        std::cerr << "WARNING: error running ogrinfo, skipping test" << std::endl;
+        return;
+    }
+
+    auto handle = FileUtils::createFile(outinfofile);
+    *handle << info;
+    FileUtils::closeFile(handle);
+
+    EXPECT_EQ(compare(infofile, outinfofile, 1), 0) \
+        << "  (" << infofile << " <-> " << outinfofile << ")";
+}
+
+uint32_t diff_geojson(const std::string& file1, const std::string& file2, int32_t ignoreLine1=-1)
```

Comment:
Comparing output of ogrinfo seems fragile to me.
Did you consider adding a reference/expected geojson file, and using OGR API to compare the one generated by the test with the reference one ? OGR_F_Equal() should help for that